cluster/src/dunit/scala/io/snappydata/cluster/SplitSnappyClusterDUnitTest.scala

```diff
@@ -263,56 +274,50 @@ class SplitSnappyClusterDUnitTest(s: String)
-      sns.sql(s"""deploy jar testsch."avro-v_1.5" '$jarPath'""")
-      assert(sns.sql("list packages").count() == 13)
-    }
-    finally {
-      sns.sql("undeploy  mongo_spark")
-      sns.sql("undeploy  mongo-spark_v1.0")
-      sns.sql("undeploy  app.mongo-spark_v1.1")
-      sns.sql("undeploy  testsch.mongo-spark_v1.2")
-      sns.sql(s"""undeploy  "testsch"."mongo-spark_v1.3" """)
-      sns.sql(s"""undeploy  testsch."mongo-spark_v1.4" """)
-      sns.sql(s"""undeploy "testsch".mongo-spark_v1.5 """)
-
-      sns.sql("undeploy  avro-v_1.0 ")
-      sns.sql("undeploy  app.avro-v_1.1")
-      sns.sql("undeploy  testsch.avro-v_1.2")
-      sns.sql(s"""undeploy "app".avro-v_1.3 """)
-      sns.sql(s"""undeploy "testsch"."avro-v_1.4" """)
-      sns.sql(s"""undeploy testsch."avro-v_1.5" """)
-      assert(sns.sql("list packages").count() == 0)
-    }
+    functionCheck(sns, "Jars/packages/UDFs not cleaned up from previous tests! ")
+    val jarPath = s"$sparkProductDir/jars/hadoop-client-2.7.7.jar"
+    sns.sql("deploy package  mongo_spark 'org.mongodb.spark:mongo-spark-connector_2.11:2.2.2'")
+    sns.sql("undeploy  mongo_spark")
+    sns.sql("deploy package mongo-spark_v1.0  'org.mongodb.spark:mongo-spark-" +
+        "connector_2.11:2.2.2'")
+    sns.sql("undeploy  mongo-spark_v1.0")
+    sns.sql("deploy package app.mongo-spark_v1.1  'org.mongodb.spark:mongo-spark-" +
+        "connector_2.11:2.2.2'")
+    sns.sql("undeploy  app.mongo-spark_v1.1")
+    sns.sql("deploy package testsch.mongo-spark_v1.2  'org.mongodb.spark:mongo-spark" +
+        "-connector_2.11:2.2.2'")
+    sns.sql("undeploy  testsch.mongo-spark_v1.2")
+    sns.sql(s"""deploy package "testsch"."mongo-spark_v1.3"  'org.mongodb.spark:mongo""" +
+          "-spark-connector_2.11:2.2.2'")
+    sns.sql(s"""undeploy  "testsch"."mongo-spark_v1.3" """)
+    sns.sql(s"""deploy package testsch."mongo-spark_v1.4"  'org.mongodb.spark:mongo""" +
+          "-spark-connector_2.11:2.2.2'")
+    sns.sql(s"""undeploy  testsch."mongo-spark_v1.4" """)
+    sns.sql(s"""deploy package "testsch".mongo-spark_v1.5  'org.mongodb.spark:mongo""" +
+          "-spark-connector_2.11:2.2.2'")
+    sns.sql(s"""undeploy "testsch".mongo-spark_v1.5 """)
+    assert(sns.sql("list packages").count() == 0)
+
+    sns.sql(s"deploy jar avro-v_1.0 '$jarPath'")
+    sns.sql("undeploy  avro-v_1.0 ")
+    sns.sql(s"deploy jar app.avro-v_1.1 '$jarPath'")
+    sns.sql("undeploy  app.avro-v_1.1")
+    sns.sql(s"deploy jar testsch.avro-v_1.2 '$jarPath'")
+    sns.sql("undeploy  testsch.avro-v_1.2")
+    sns.sql(s"""deploy jar "app".avro-v_1.3 '$jarPath'""")
+    sns.sql(s"""undeploy "app".avro-v_1.3 """)
+    sns.sql(s"""deploy jar "testsch"."avro-v_1.4" '$jarPath'""")
+    sns.sql(s"""undeploy "testsch"."avro-v_1.4" """)
+    sns.sql(s"""deploy jar testsch."avro-v_1.5" '$jarPath'""")
+    sns.sql(s"""undeploy testsch."avro-v_1.5" """)
+
+    assert(sns.sql("list packages").count() == 0)
+
```

Comment:
These deploy jar related tests need to be moved to some other relevant dunit test as SplitSnappClusterDUniteTest.scala is for smart connector tests.

Reply:
Agree. We'll track it for post-release.